### DISCUSSION TITLE
Simple Storage

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#823] [Docs]
<meta url="https://github.com/johnlindquist/kit/discussions/823">
<meta id="D_kwDOEu7MBc4AQA6S">
<meta sectionIndex="3">
<meta title="Simple Storage">
<meta section="Files and Data">
<meta i="4">    
<meta path="docs/simple-storage">

## Read and Write Using the `db` Helper

The `db` helpers reads/writes to json files in the `~/.kenv/db` directory. It's meant as a simple wrapper around common json operations.

```js
// Menu: Database Read/Write Example
// Description: Add/remove items from a list of fruit

let fruitDb = await db(["apple", "banana", "orange"])

// This will keep prompting until you hit Escape
while (true) {
  let fruitToAdd = await arg(
    {
      placeholder: "Add a fruit",
      //allows to submit input not in the list
      strict: false,
    },
    fruitDb.items
  )

  fruitDb.items.push(fruitToAdd)
  await fruitDb.write()

  let fruitToDelete = await arg(
    "Delete a fruit",
    fruitDb.items
  )

  fruitDb.items = fruitDb.items.filter(
    fruit => fruit !== fruitToDelete
  )

  await fruitDb.write()
}
```

[Open db-store in Script Kit](https://scriptkit.com/api/new?name=db-store&url=https://gist.githubusercontent.com/johnlindquist/a7cda43e196f6b6e38e4c66cba8cdb74/raw/8d93dc14970bac042763cb86b30456b32ba5fab7/db-store.js")

**Comment by peterroe** [comment]
Nice feature!

**Comment by frontendwizard** [comment]
I notice the types don't really allow you to type the db. Reading the code, I found this:

```
export declare let db: (key: any, defaults?: any, fromCache?: boolean) => Promise<Low & any>;
```

I suppose this type should change to support a generic that describe the type of things that are going to live on this db. It's really clear to me what `key` really means in this context. from this script it looks like `key` is actually the initial items on the db. Do all of these items get inserted into a `items` array?

**Reply by frontendwizard** [reply]
oh, ts type is just misleading. If I understood correctly, this method also has an overload that allow you to use it without a key param, just passing the defaults and that gets the content into an items array. I'll open an issue on this.

**Reply by frontendwizard** [reply]
#859
